opti/tools/reduce.py

```diff
@@ -0,0 +1,271 @@
+from logging.handlers import WatchedFileHandler
+import numpy as np
+import pandas as pd                                                     
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+
+from opti import Problem
+from opti.constraint import Constraints, LinearEquality
+from opti.objective import Objectives
+from opti.parameter import Categorical, Parameters
+from sympy import Matrix
+
+
+#TODO: Wie geht man mit diskreten inputs um? --> einfach domain beibehalten?
```

Comment:
Discrete variable typically don't enter mixture constraints. 
I would propose to only consider those continuouss inputs for reduction that appear in the linear equality. If a discrete or categorical input is part of a linear equality, then an exception is raised.